maintainer/docker_build.sh

```diff
@@ -1,15 +1,21 @@
 #!/bin/sh
 
-# execute and output a command
-function cmd {
-    echo ">$1"
-    eval $1
+abort()
+{
+    echo "An error occurred. Exiting..." >&2
+    echo "Command that failed: $BASH_COMMAND" >&2
+    exit 1
 }
 
-cmd "cd docker/${CI_JOB_NAME%:*}"
+trap 'abort' 0
+set -ex
```

Comment:
I don't like your solution. Why don't you just leave everything as-is and change
```bash
function cmd {
    echo ">$1"
    eval $1
    if [ "$?" != "0" ]; then
        echo "An error occurred. Exiting..." >&2
        exit 1
    fi
}
```

Reply:
You can push to this branch if you want to have it exactly like your suggestion. I did not really introduce sth new here. We handle error exactly the same in the build scripts for the espresso repository. This syntax has the advantage that it does not pollute the script.

Reply:
`set -e` doesn't always have the expected behavior (I think you even dug up a blog post on that a while ago), the espresso build script is a horrible mess, and I can't push to your branch because I don't have maintainer rights.